features/FEATURE_BLE/targets/TARGET_STM/TARGET_NUCLEO_WB55RG/stm32wb_HCIDriver.cpp

```diff
@@ -0,0 +1,902 @@
+
+        bleCmdBuf = BleCmdBuffer;
+        sysCmdBuf = SystemCmdBuffer;
+        evtPool = EvtPool;
+        sysSpareEvtBuf = SystemSpareEvtBuffer;
+        bleSpareEvtBuf = BleSpareEvtBuffer;
+        aclDataBuffer = HciAclDataBuffer;
+    }
+
+    virtual ~TransportDriver() { }
+
+    /**
+     * @see CordioHCITransportDriver::initialize
+     */
+    virtual void initialize()
+    {
+        init_debug();
+        stm32wb_reset();
+        transport_init();
+    }
+
+    /**
+     * @see CordioHCITransportDriver::terminate
+     */
+    virtual void terminate() { }
+
+    /**
+     * @see CordioHCITransportDriver::write
+     */
+    virtual uint16_t write(uint8_t type, uint16_t len, uint8_t *pData)
+    {
+        return mbox_write(type, len, pData);
+    }
+
+private:
+    void transport_init(void) {
+        TL_MM_Config_t tl_mm_config;
+        TL_BLE_InitConf_t tl_ble_Config;
+        /* STM32WB offers a System Channel HCI interface for
+           offering system services, with proprietary commands.
+           System Channel must be used as well for starting up
+           BLE service so we need to initialize it. */
+        SHCI_TL_HciInitConf_t shci_init_config;
+
+        /**< Reference table initialization */
+        TL_Init();
+
+        /**< System channel initialization */
+        shci_init_config.p_cmdbuffer = (uint8_t*)sysCmdBuf;
+        shci_init_config.StatusNotCallBack = syscmd_status_not;
+        shci_init(sysevt_received, (void*) &shci_init_config);
+
+        /**< Memory Manager channel initialization */
+        tl_mm_config.p_BleSpareEvtBuffer = bleSpareEvtBuf;
+        tl_mm_config.p_SystemSpareEvtBuffer = sysSpareEvtBuf;
+        tl_mm_config.p_AsynchEvtPool = evtPool;
+        tl_mm_config.AsynchEvtPoolSize = POOL_SIZE;
+        TL_MM_Init(&tl_mm_config);
+
+        TL_Enable();
+
+        /*  At this stage, we'll need to wait for ready event,
+         *  passed thru TL_SYS_EvtReceived */
+        if(!sysevt_wait()) {
+            tr_debug("ERROR booting WB controler\r\n");
+            return;
+        }
+
+        // TO DO : check if we need to disable LPM
+        // requires to import as well all lpm driver
+        tl_ble_Config.p_AclDataBuffer = aclDataBuffer;
+        tl_ble_Config.IoBusAclDataTxAck = acl_data_ack;
+        tl_ble_Config.p_cmdbuffer = (uint8_t *)bleCmdBuf;
+        tl_ble_Config.IoBusEvtCallBack = evt_received;
+        TL_BLE_Init(&tl_ble_Config);
+
+
+        /* Now start BLE service on firmware side, using Vendor specific
+         * command on the System Channe
+         */
+        stm32wb_start_ble();
+    }
+
+     uint16_t mbox_write(uint8_t type, uint16_t len, uint8_t *pData) {
+        // Note:
+        // 01 command
+        // type 02 ACL DATA not yet supported
+        // type 03 SCO Voice
```

Comment:
this seems out of place, also why not enum?

Reply:
I agree - this comes from the reference Cube software that will most probably evolve over time ... so the enum may soon be introduced and I will use it